component/olm.jsonnet

```diff
@@ -170,14 +184,38 @@ local patchManifests = function(file)
     file.contents.metadata.namespace == 'cilium'
   ) then
     null
+  else if (
+    !has_csv &&
+    file.contents.kind == 'OperatorGroup' &&
+    file.contents.metadata.namespace == 'cilium'
+  ) then
+    null
+  else if (
+    file.contents.kind == 'Role' &&
+    file.contents.metadata.namespace == 'cilium' &&
+    file.contents.metadata.name == metadata_name_map[params.release].OlmRole
+  ) then
+    file {
+      contents+: {
+        rules: [
+          if r.apiGroups == [ '' ] && r.resources == [ 'events' ] then
+            r {
+              verbs+: [ 'patch' ],
+            }
+          else
+            r
+          for r in super.rules
+        ],
+      },
+    }
```

Comment:
I get what this does, but I lack the context to know why it's needed. I'm curious.

Reply:
It's needed because the upstream manifests are missing this permission which the olm operator tries to use. Since we don't really need the events, it's mostly cosmetic, and may reduce the amount of log spam in the olm operator pod (since there was a log message for each time the operator tries to patch an event but isn't allowed to do so)